src/main/java/core/basesyntax/service/impl/operation/SupplyOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.Fruit;
+
+public class SupplyOperation implements OperationHandler {
+    private final FruitStorageDao fruitStorageDao;
+
+    public SupplyOperation(FruitStorageDao fruitStorageDao) {
+        this.fruitStorageDao = fruitStorageDao;
+    }
+
+    @Override
+    public boolean doOperation(Fruit fruit, int quantity) {
+        if (fruit == null) {
+            throw new RuntimeException("Fruit can't be null");
+        }
+        if (quantity < MIN_FRUIT_QUANTITY) {
+            throw new RuntimeException("Quantity " + quantity + " can't be negative");
+        }
+        return fruitStorageDao.add(fruit, quantity);
+    }
+}
```

Comment:
The class `SupplyOperation` should be placed in the `strategy` package according to the checklist, but it's currently in the `impl.operation` package. Consider refactoring the package structure.